### ISSUE TITLE
Find and use CC-licensed images

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/ideas#109]
### Feature Description

A number of our authors have asked us questions about how they can find and use already-existing openly-licensed images in book projects they're writing. We often show them various Creative Commons search tools, like the [classic search engine](https://search.creativecommons.org/) and their newish [beta search tool]( https://ccsearch.creativecommons.org/) ([more info from WP Tavern](https://wptavern.com/creative-commons-new-search-tool-is-now-in-beta-pulls-cc-images-from-multiple-sources)). Some users have managed to use these tools, but finding, saving, importing, and attributing images is still a fairly manual and tedious process with a bit of a technical barrier to entry. 

At the same time, I'm also aware of a certain class of WP plugins that allow users to find and insert images from well-established image hosting sites with good APIs. The most prominent of the free ones are probably [WP Inject](https://wordpress.org/plugins/wp-inject/) or [PhotoDropper](https://wordpress.org/plugins/photo-dropper/) or [Unsplash](https://wordpress.org/plugins/instant-images/), and I understand there are a handful of premium plugins that do something similar (like [Free Image Search](https://codecanyon.net/item/free-image-search-creative-commons-image-search/15723831), to give one example).

In short: we'd like to allow Pressbooks authors to search as much of the commons as possible (difficult, because it's so distributed, we know, but ideally some combination of Flickr, Wikimedia Commons, Google Images, and perhaps Pixabay or other widely-used free stock image sites--a list of these could be provided or curated by members of the community) for openly licensed image content and to add these visual resources to their books/media libraries with the proper metadata & attribution information as easily as possible within the WYSIWYG editor.

### Feature Use Case

I imagine that this feature would be of use to all Pressbooks authors with an interest in incorporating visual resources into a book they're authoring.

**Comment by SteelWagstaff** [comment]
@bdolor -- I think you and Alex are working on citations for core Pressbooks at present. Is that correct? I missed the call yesterday, but heard a second-hand report. Is there any way to connect that work with what we're interested in here?

**Comment by bdolor** [comment]
Hi Steel, I have a note in my notebook from 2014 or so when someone from SUNY/CUNY approached me with a find and use CC images idea. I thought it was a great idea then and think it's a great idea now and I would personally love to work on something like this. What Alex and I built, however, was in response to requests through the Open Textbook Project and ends up being an attribution builder, of sorts. The feature provides a integrated way (with the media manager) to input attribution fields (TASL) and a standardized way to output attribution statements. The searching and importing piece, however, was not in scope. More info here: https://github.com/pressbooks/pressbooks/pull/1287 

What we're working on now is a glossary for PB — we hope to have a PR ready at the end of the week. I  agree that this searching and importing of CC licensed images would be a useful feature to have.

**Comment by SteelWagstaff** [comment]
Thanks Brad. The PR description/details is very helpful, and yes, had heard about your glossary work and am very excited to see/use/test it. Thanks for all of your excellent contributions!

**Comment by SteelWagstaff** [comment]
See https://creativecommons.org/2020/03/16/version-one-of-the-cc-catalog-api/ + https://opensource.creativecommons.org/blog/entries/2019-07-24-cc-search-wp-plugin/ + https://github.com/lenasterg/ls-wp-ccsearch for some promising developments on this front!

**Comment by SteelWagstaff** [comment]
Still a good idea 😂

**Comment by arzola** [comment]
This issue has been automatically closed as part of an organization-wide backlog review.

**Reason:** This issue was created 7y 7m ago with no updates in the last 1y 2m.

If this is still relevant, please feel free to reopen it with updated context. We appreciate your contributions and want to ensure our issue tracker stays actionable.

**Learn more:** https://pressbooks.community/t/announcement-automatic-closure-of-old-github-tickets/4542
